<launch>
    <arg name="binning_on" default="false"/>
    <arg name="software_binning_on" default="false"/>
    <arg name="software_binning_level" default="2"/>
    <arg name="triggered_on" default="true"/>
    <arg name="auto_exposure_control" default="false"/>
    <arg name="auto_gain_control" default="false"/>
    <arg name="high_dynamic_range_on" default="false"/>
    <arg name="expose_us" default="13000"/>
    <arg name="frame_rate" default="20.0"/>

    <node pkg="bluefox" name="bluefox_node0" type="bluefox_node" output="screen">
        <param name="binning_on" type="bool" value="$(arg binning_on)"/>
        <param name="software_binning_on" type="bool" value="$(arg software_binning_on)"/>
        <param name="software_binning_level" type="int" value="$(arg software_binning_level)"/>
        <param name="triggered_on" type="bool" value="$(arg triggered_on)"/>
        <param name="aec_on" type="bool" value="$(arg auto_exposure_control)"/>
        <param name="agc_on" type="bool" value="$(arg auto_gain_control)"/>
        <param name="hdr_on" type="bool" value="$(arg high_dynamic_range_on)"/>
	    <param name="expose_us" type="int" value="$(arg expose_us)"/>
        <param name="frame_rate" type="double" value="$(arg frame_rate)"/>
        <param name="cam_id" type="int" value="0"/>
        <param name="serial" type="string" value="25003728"/>
    </node>

    <node pkg="bluefox" name="bluefox_node1" type="bluefox_node" output="screen">
        <param name="binning_on" type="bool" value="$(arg binning_on)"/>
        <param name="software_binning_on" type="bool" value="$(arg software_binning_on)"/>
        <param name="software_binning_level" type="int" value="$(arg software_binning_level)"/>
        <param name="triggered_on" type="bool" value="$(arg triggered_on)"/>
        <param name="aec_on" type="bool" value="$(arg auto_exposure_control)"/>
        <param name="agc_on" type="bool" value="$(arg auto_gain_control)"/>
        <param name="hdr_on" type="bool" value="$(arg high_dynamic_range_on)"/>
	    <param name="expose_us" type="int" value="$(arg expose_us)"/>
        <param name="frame_rate" type="double" value="$(arg frame_rate)"/>
        <param name="cam_id" type="int" value="1"/>
        <param name="serial" type="string" value="25001575"/>
    </node>

    <node pkg="bluefox" name="bluefox_node2" type="bluefox_node" output="screen">
        <param name="binning_on" type="bool" value="$(arg binning_on)"/>
        <param name="software_binning_on" type="bool" value="$(arg software_binning_on)"/>
        <param name="software_binning_level" type="int" value="$(arg software_binning_level)"/>
        <param name="triggered_on" type="bool" value="$(arg triggered_on)"/>
        <param name="aec_on" type="bool" value="$(arg auto_exposure_control)"/>
        <param name="agc_on" type="bool" value="$(arg auto_gain_control)"/>
        <param name="hdr_on" type="bool" value="$(arg high_dynamic_range_on)"/>
	    <param name="expose_us" type="int" value="$(arg expose_us)"/>
        <param name="frame_rate" type="double" value="$(arg frame_rate)"/>
        <param name="cam_id" type="int" value="2"/>
        <param name="serial" type="string" value="25003716"/>
    </node>

    <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure" output="screen">
    </node>

    <node pkg="rqt_image_view" type="rqt_image_view" name="rqt_image_view" output="screen">
    </node>
</launch>
